docs/source/tutorials/token_classification/entity_recognition_training.ipynb

```diff
@@ -0,0 +1,737 @@
+    "\n",
+    "sentences, mask = filter_sentence(sentences) \n",
+    "given_words = [words for m, words in zip(mask, given_words) if m] \n",
+    "given_labels = [labels for m, labels in zip(mask, given_labels) if m] "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "50c496b8",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Number of sentences: 20718\n",
+      "Eu rejects German call to boycott British lamb.\n"
+     ]
+    }
+   ],
+   "source": [
+    "print('Number of sentences: %d' % len(sentences)) \n",
+    "print(sentences[0]) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0fb1b2b0",
+   "metadata": {},
+   "source": [
+    "## 3. Train Models using Cross-Validation \n",
+    "\n",
+    "To identify potential label errors in the training dataset, we compute the out-of-sample predicted probabilities using cross-validation. We first partition the dataset into k-folds: "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "797df997",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "'folds/' already exists, skipping...\n"
+     ]
+    }
+   ],
+   "source": [
+    "lines = [[]] \n",
+    "for filepath in filepaths: \n",
+    "    for line in open(filepath) : \n",
+    "        if len(line) == 0 or line.startswith('-DOCSTART') or line[0] == '\\n':\n",
+    "            if len(lines[-1]) > 0: \n",
+    "                lines.append([]) \n",
+    "        else: \n",
+    "            lines[-1].append(line) \n",
+    "        \n",
+    "lines = lines[:-1] \n",
+    "lines = [line for m, line in zip(mask, lines) if m] \n",
+    "\n",
+    "k = 5 \n",
+    "indicies = create_folds(lines, k=k) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "304653aa",
+   "metadata": {},
+   "source": [
+    "We train one model for each training/testing pair: "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "id": "6c877f7e",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Model 0 already exists, skipping...\n",
+      "Model 1 already exists, skipping...\n",
+      "Model 2 already exists, skipping...\n",
+      "Model 3 already exists, skipping...\n",
+      "Model 4 already exists, skipping...\n"
+     ]
+    }
+   ],
+   "source": [
+    "for i in range(k): \n",
+    "    if os.path.exists('folds/fold%d/model/' % i): \n",
+    "        print('Model %d already exists, skipping...' % i) \n",
+    "    else: \n",
+    "        os.system(\n",
+    "            \"python3 run_ner.py --data_dir=folds/fold%d --bert_model=bert-base-cased \" \\\n",
+    "            \"--task_name=ner --output_dir=folds/fold%d/model --max_seq_length=256 \" \\\n",
+    "            \"--do_train --num_train_epochs 10 --warmup_proportion=0.1\" % (i, i)) "
+   ]
```

Comment:
Is invoking the training like this better than just adding a function inside `run_ner.py` that is imported into this notebook and has the same effect?

Reply:
Between each fold of cross-validation (inside the loop), let's print status update:

`print(f"Training model on fold {i} (out of {k}) of cross-validation ...")`

and ensure something is printed when the model is saved for each fold.

Reply:
this is definitely possible but will take a while to implement, so I add a TODO there and will revisit later